packages/starknet-snap/src/recoverAccounts.ts

```diff
@@ -33,7 +33,8 @@ export async function recoverAccounts(params: ApiParams) {
         address: contractAddress,
         signerPubKey: signerPublicKey,
         upgradeRequired,
-      } = await getCorrectContractAddress(network, publicKey);
+        deployRequired,
+      } = await getCorrectContractAddress(network, publicKey, state);
```

Comment:
i think we dont need to put the state in the function, as if we wanna get getEtherErc20Token address, if the state is missing, then ur address cant found (likely not possible), i will recommend just use the hardcode address in the getCorrectContractAddress